vignettes/datatable-joins.Rmd

```diff
@@ -569,6 +569,39 @@ ProductReceivedProd2[ProductSalesProd2,
                      nomatch = NULL]
 ```
 
+### 4.1 Handling Column Name Behavior in Non-Equi Joins
+
+When performing non-equi joins (<, >, <=, >=), column names are assigned as follows:
+
+- The left operand (`x` column) determines the column name in the result.
+- The right operand (`i` column) contributes values but does not retain its original name.
+- By default, `data.table` does not retain the `i` column used in the join condition unless explicitly requested.
+
+In non-equi joins, the left side of the operator (e.g., x_int in x_int >= i_int) must be a column from x, while the right side (e.g., i_int) must be a column from i. Non-equi joins do not support arbitrary expressions.
+For example, on = .(x_int >= i_int) is valid, but on = .(x_int >= i_int + 1) is not valid.
+
+```{r}
+x <- data.table(x_int = 2:4, lower = letters[1:3])
+i <- data.table(i_int = c(2, 4, 5), UPPER = LETTERS[1:3])
+x[i, on = .(x_int >= i_int)]
+```
+Key Takeaways:
+- The name of the output column (x_int) comes from x, but the values come from i_int in i.
+- The last row contains NA because no rows in x match the last row in i (UPPER = "C").
+- Multiple rows in x are returned to match the first row in i with UPPER = "A".
```

Comment:
```suggestion
- The name of the output column (`x_int`) comes from `x`, but the values come from `i_int` in `i`.
- The last row contains `NA` because no rows in `x` match the last row in `i` (`UPPER == "C"`).
- Multiple rows in `x` are returned to match the first row in `i` with `UPPER == "A"`.
```